editoast/src/views/timetable/similar_trains/trains_traffic.rs

```diff
@@ -0,0 +1,361 @@
+use crate::client::trains_traffic::JsonTrainTraffic;
+use crate::error::Result;
+use crate::views::timetable::similar_trains::Waypoint;
+use crate::views::timetable::similar_trains::squash_successive_waypoints;
+use arcstr::ArcStr;
+use chrono::DateTime;
+use chrono::Months;
+
+use chrono::Utc;
+use itertools::Itertools;
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::collections::HashSet;
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct TrainTraffic {
+    pub id: usize,
+    pub train_name: String,
+    pub start_time: DateTime<Utc>,
+    pub operational_points: Vec<Waypoint>,
```

Comment:
```suggestion
    pub past_train: past_train::PastTrain,
    pub train_name: String,
    pub start_time: DateTime<Utc>,
```

it's the semantics of this type: a past train that can be explored with additional information for the response